kernel/src/fs/procfs/pid/mem.rs

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use align_ext::AlignExt;
+use ostd::{
+    mm::{io_util::HasVmReaderWriter, UFrame},
+    task::disable_preempt,
+};
+
+use crate::{
+    fs::{
+        procfs::template::{FileOps, ProcFileBuilder},
+        utils::{Inode, InodeMode},
+    },
+    prelude::*,
+    Process,
+};
+
+/// Represents the inode at either `/proc/[pid]/mem` or `/proc/[pid]/task/[tid]/mem`.
+pub struct MemFileOps(Arc<Process>);
+
+impl MemFileOps {
+    pub fn new_inode(process_ref: Arc<Process>, parent: Weak<dyn Inode>) -> Arc<dyn Inode> {
+        ProcFileBuilder::new(Self(process_ref))
+            .parent(parent)
+            // Reference: <https://github.com/torvalds/linux/blob/0ff41df1cb268fc69e703a08a57ee14ae967d0ca/fs/proc/base.c#L3341>
+            .mode(InodeMode::from_bits_truncate(0o600))
+            .build()
+            .unwrap()
+    }
+
+    fn access_at<F>(&self, offset: usize, len: usize, mut op: F) -> Result<usize>
+    where
+        F: FnMut(UFrame, usize) -> Result<()>,
+    {
+        let range = offset.align_down(PAGE_SIZE)..(offset + len).align_up(PAGE_SIZE);
+
+        let vmar_guard = self.0.lock_root_vmar();
+        let vmar = vmar_guard.as_ref().ok_or(Error::new(Errno::ENOENT))?;
```

Comment:
Wrong error code. Let's figure out the correct error code by performing experiments and properly write the error message here:

```
> cat /proc/351166/stat
351166 (a) Z 351165 351165 349720 34826 351165 4227148 38 0 0 0 0 0 0 0 20 0 1 0 13823004 0 0 18446744073709551615 0 0 0 0 0 0 0 0 0 1 0 0 17 15 0 0 0 0 0 0 0 0 0 0 0 0 0
> sudo cat /proc/351166/mem
cat: /proc/351166/mem: No such process
```

```suggestion
        let vmar = vmar_guard.as_ref().ok_or_else(|| Error::with_message(Errno::ESRCH, "the process has exited"))?;
```